annotations/builder/src/main/java/io/sundr/builder/internal/functions/ToMethod.java

```diff
@@ -744,12 +740,20 @@ public List<Method> apply(final Property property) {
 
             addSingleItemAtIndex = new MethodBuilder(addSingleItemAtIndex).withParameters(parameters).editBlock()
                 .withStatements(createAddToDescendants("addTo", descendants, true),
-                    new StringStatement("return (" + returnType + ")this;"))
```

Comment:
There is a method available caleld `createAddToDescendantsFallback` maybe this should be used instead?
It seems as `createAddToDescendantsFallback` was meant to do the same thing but somehow got forgotten.

Reply:
> There is a method available caleld createAddToDescendantsFallback maybe this should be used instead?

To clarify there are two things happening.  One is usage of builderOf, the other is the actual operation.  createAddToDescendantsFallback is written to do both, so it's not applicable to the other operations - indexed addTo, setTo, remove.

Note that not everywhere createAddOrSetIndex is a fallback case - the same logic applies to both fallback https://github.com/sundrio/sundrio/pull/375/files#diff-f95a091f7367292c3d8e6c10eb03b6411bfe67573270d10bb9393f947dc5ec54R755 and regular logic https://github.com/sundrio/sundrio/pull/375/files#diff-f95a091f7367292c3d8e6c10eb03b6411bfe67573270d10bb9393f947dc5ec54R773 - this does ensure that consistent indexing is used across the methods, which is also called out in #374